<a href="https://colab.research.google.com/github/An1ndya/IIT-Kharagpur/blob/main/ML01_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

1. Write a program to compute and print the taxi fare based on the following chart. Total number of Kilometers 
traveled will be input by the user.
First 12 KM: Rs. 100/-
Next 4 KM: Rs. 8 / KM
Next 4 KM: Rs. 6 / KM
Above 20 KM: Rs. 5 / KM


In [ ]:

x=int(input())

y= 100

if   x>20 : y+= 5*(x-20)+4*6+4*8
elif x>16 : y+= 6*(x-16)+4*8
elif x>12 : y+= 8*(x-12)
elif x==0 : y=0

print(y)

13
108


2. Acceleration due to gravity of a celestial object of mass M and radius R is given by
𝑔𝑔 = 𝐺𝐺
𝑀𝑀
𝑅𝑅2
Calculate and print the values of g’s for the earth and moon, given that
● Mass of the earth = 5.972 × 1024 kg
● Radius of the earth = 6361 km
● Mass of the moon = 7.35 x 1022 kg
● Radius of the moon = 1737 km
● Assume the value of Newton’s constant G as 6.67408×10−11 m3
⋅kg−1
⋅s
−2


In [ ]:
def gravity(m,r):
  gc = 6.67408*10e-11
  return gc*m/(r*r)
mE = 5.972 * 10e24
rE = 6361 * 10e3
mM = 7.35 * 10e22
rM = 1737 * 10e3
print("Earth's g:", gravity(mE,rE))
print("Moon's g:", gravity(mM,rM))

Earth's g: 9.850548553554308
Moon's g: 1.6258448896962683


3. Write a program to find the sum of the following series for a given value of n.
𝑆𝑆 = 1 − 1
3 +
1
32 − 1
33 + ⋯ (−1)𝑛𝑛 1
3𝑛𝑛
Your program should take n as input from the user and print the sum.

In [ ]:
n =int(input())
s=0
a=1
r=3
s=sn=s0=1
for i in range(1,n+1):
  sn /= -r
  s+=sn
print(s)

5
0.7489711934156381


4. An Armstrong number is the number which is the sum of the cubes of all its units, tens and hundred digits, etc. 
For example, for a three-digit number 153,
153 = 1 ∗ 1 ∗ 1 + 5 ∗ 5 ∗ 5 + 3 ∗ 3 ∗ 3 = 1 + 125 + 27 = 153
For a four-digit number 1634,
1634 = 1 ∗ 1 ∗ 1 ∗ 1 + 6 ∗ 6 ∗ 6 ∗ 6 + 3 ∗ 3 ∗ 3 ∗ 3 + 4 ∗ 4 ∗ 4 ∗ 4 = 1 + 1296 + 81 + 256 = 1634
Write a Python Program to check if the user entered number is Armstrong number or not.

In [ ]:
n =int(input())
k=n
sum=0
c=0
while k > 0:
  k = int(k/10)
  c+=1
k=n
while k > 0:
  mod = k%10
  k = int(k/10)
  sum += mod**c
if sum==n : print(n," is an Armstrong number.")
else : print(n," is Not an Armstrong number.")

153
153  is an Armstrong number.


5. A number is called Harshad number (also called Niven number) if the number is divisible by the sum of its digits. 
For example, 210 is a Harshad number because 210 is divisible by the sum of its digits (2 + 1 + 0 = 3). Write a 
program which will print the first 10 Harshad numbers with n-digits. The number n will be known at the time of 
running your program. 

In [ ]:
def isNiven(n):
  k=n
  sum=0
  while k>0:
    sum+=k%10
    k=int(k/10)
  return n%sum==0

n=int(input())
c=0
a=10**(n-1)
max=9**n
while c<10 and a<=max:
  if isNiven(a) :
    print(a)
    c+=1
  a+=1

2
10
12
18
20
21
24
27
30
36
40


6. Write a Python Program to find whether a string or number is palindrome or not, and print the decision as output.

In [ ]:
str = input()
n = len(str)
isPalindrome = True
for i in range(0,int(n/2)):
  if str[i]!=str[n-1-i]:
    isPalindrome = False  
    break
    
if isPalindrome : print(str, "is a Palindrome")
else: print(str, "is Not a palindrome")


123321
123321 is a Palindrome


7. Evaluate the expression of Tp, and print the result.
𝑇𝑇𝑝𝑝 = 𝑇𝑇𝑠𝑠
�𝑅𝑅𝑠𝑠�1−∝
𝜎𝜎
2𝐷𝐷
Where, 𝛼𝛼 = 0.306, 𝑇𝑇𝑇𝑇 = 6.96 × 108 𝑚𝑚, 𝑅𝑅𝑅𝑅 = 6.96 × 108 𝑚𝑚, 𝐷𝐷 = 1.496 × 1011 𝑚𝑚 𝑎𝑎𝑎𝑎𝑎𝑎 𝜎𝜎 = 1.2

In [1]:
import math
a  =.306
Ts = 6.96*10e8
Rs = 6.96*10e8
D  = 1.496*10e11
s  = 1.2

Tp = math.sqrt(Rs*math.sqrt((1-a)/s)/(2*D))

print(Tp)


0.04205991409322086


8. Symmetric matrix is a square matrix which is equal to its transpose. If 𝐴𝐴[][] is a square matrix with (𝑛𝑛 𝑥𝑥 𝑛𝑛) order, 
then this matrix is said to be symmetric if every element at 𝑖𝑖
𝑡𝑡ℎ row and 𝑗𝑗𝑡𝑡ℎ column is equal to element at 𝑗𝑗𝑡𝑡ℎ row 
and 𝑖𝑖
𝑡𝑡ℎ column, that is, 𝐴𝐴[𝑖𝑖][𝑗𝑗] == 𝐴𝐴[𝑗𝑗][𝑖𝑖]. Your program should take the input matrix from the user, display it 
and check whether the matrix is symmetric or not

In [13]:
r = int(input("Enter the number of rows: "))
c = int(input("Enter the number of columns: "))  
# Initialize matrix
mat= []
print("Enter the entries rowwise:")
for i in range(r):         
    a =[]
    for j in range(c):    
         a.append(int(input()))
    mat.append(a)

isSymmetric = True

if r==c:
  for i in range(0,r):
    for j in range(0,c):
      if mat[i][j]!=mat[j][i]:
        isSymmetric = False
        break
else:
  isSymmetric = False

if isSymmetric : print(mat, "is a Symmetric matrix")
else : print(mat, "is Not a Symmetric matrix")

Enter the number of rows: 2
Enter the number of columns: 2
Enter the entries rowwise:
1
2
2
1
[[1, 2], [2, 1]] is a Symmetric matrix


9. Saddle point of a matrix is an element in the matrix which is smallest in its row and largest in its column. For 
example, in the following matrix, 7 is the saddle point at (2,2).
6 3 1
9 7 8
2 4 5
Write a program to find all the saddle point in a given matrix

In [25]:
import sys
r = int(input("Enter the number of rows: "))
c = int(input("Enter the number of columns: "))  
# Initialize matrix
mat= []
print("Enter the entries rowwise:")
for i in range(r):         
  a =[]
  for j in range(c):    
        a.append(int(input()))
  mat.append(a)

rowMinIndexofC = []
for i in range(r): 
  rowMinIndexofC.append(-1)
  min= int(sys.maxsize)
  for j in range(c):    
    if mat[i][j]<min: 
      min = mat[i][j]
      rowMinIndexofC[i]=j

colMaxIndexofR = []
for j in range(c): 
  colMaxIndexofR.append(-1)
  max= -int(sys.maxsize)
  for i in range(r):    
    if mat[i][j]>max: 
      max = mat[i][j]
      colMaxIndexofR[j]=i
count = 1
for i in range(r):         
  for j in range(c): 
    if i==colMaxIndexofR[j] and j == rowMinIndexofC[i] :
      print("saddle point ",count, ": (", i+1,j+1, ")")
      count+=1
      

Enter the number of rows: 3
Enter the number of columns: 3
Enter the entries rowwise:
6
3
1
9
7
8
2
4
5
saddle point  1 : ( 2 2 )


10. A ball is released from a height of h meters. Each time it bounces on the floor, its velocity becomes halved. Write 
a program, which takes the value of h and then prints the total distance traversed by the ball when it touches the 
ground for the nth time. Assume that the value of acceleration due to gravity, g is 9.8 m/sec2
.

In [33]:
import math
h=float(input("value of h: "))
n=int(input("value of n: "))
g= 9.8
d= 0.0
v= 0.0
if n>0 : d=h
if n>1 :
  #v = math.sqrt(2*g*h)
  #h=h/4 as v=v/2
  for i in range(2,n+1):
    h=h/4
    d+=2*h

print("distance: ",d)

value of h:9.8
value of n:4
distance:  16.23125
